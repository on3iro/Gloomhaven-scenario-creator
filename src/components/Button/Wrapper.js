/**
 * Styled container for Button
 *
 * @namespace Button.Wrapper
 * @memberOf Button
  */

import styled from 'styled-components';


const Wrapper = styled.div`
  display: inline-block;
`;

export default Wrapper;
