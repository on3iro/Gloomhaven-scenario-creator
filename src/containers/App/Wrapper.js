/**
  * Styled Wrapper for the whole application
  *
  * @namespace App.Wrapper
  * @memberOf App
  */

import styled from 'styled-components';


const Wrapper = styled.div`
  height: 100%;
  min-height: 100%;
  display: flex;
  flex-direction: column;
`;

export default Wrapper;
