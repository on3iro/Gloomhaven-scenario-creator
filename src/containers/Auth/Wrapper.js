/**
  * Renders a styled div for the HomePage
  *
  * @namespace HomePage.Wrapper
  * @memberOf HomePage
  */

import styled from 'styled-components';


const Wrapper = styled.div`
  margin: 0 auto;
  width: 50%;
`;

export default Wrapper;
