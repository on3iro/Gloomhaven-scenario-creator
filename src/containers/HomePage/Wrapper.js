import styled from 'styled-components';


const Wrapper = styled.section`
  width: 50%;
  margin: 0 auto;
`;

export default Wrapper;
