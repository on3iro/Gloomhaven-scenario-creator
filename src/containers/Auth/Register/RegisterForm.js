/**
  * The user registration form
  *
  * @namespace RegisterPage.RegisterForm
  * @memberOf Register
  */

import React, { Component } from 'react';

import H1 from 'components/H1';
import Button from 'components/Button';
import Input from 'components/Input';


class RegisterForm extends Component {
  /**
    * Class component RegisterForm
    *
    * @class RegisterForm
    * @namespace RegisterForm
    * @memberOf RegisterPage.RegisterForm
    */

  constructor(props) {
    super(props);

    this.state = {
      email: '',
      password: '',
      confirm: '',
    };
  }

  handleInputChange = (event) => {
    /**
      * Handles changes on input elements.
      * Sets the value of the respective input.
      *
      * @method handleInputChange
      * @memberOf RegisterPage.RegisterForm.RegisterForm
      * @namespace handleInputChange
      * @param {Object} event - Triggered event
      * @returns {undefined}
      */

    const target = event.target;
    const value = target.type === 'checkbox' ? target.checked : target.value;
    const name = target.name;

    this.setState({
      [name]: value,
    });

    return undefined;
  }

  render() {
    return (
      <form onSubmit={() => {}}>
        <H1>Register</H1>
        <Input
          name="email"
          type="text"
          placeholder="Email"
          value={this.state.email}
          onChange={this.handleInputChange}
        />
        <Input
          name="password"
          type="password"
          placeholder="Password"
          value={this.state.password}
          onChange={this.handleInputChange}
        />
        <Input
          name="confirm"
          type="password"
          placeholder="Confirm Password"
          value={this.state.confirm}
          onChange={this.handleInputChange}
        />
        <Button submit>Register Now</Button>
      </form>
    );
  }
}

export default RegisterForm;
