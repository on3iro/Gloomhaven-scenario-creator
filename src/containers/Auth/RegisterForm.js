/*
  * The user registration form
  *
  * @namespace RegisterPage.RegisterForm
  * @memberOf Register
  */

import React, { PropTypes } from 'react';
import { Field, reduxForm } from 'redux-form';

import H1 from 'components/H1';
import Button from 'components/Button';

import RenderField from './RenderField';


export const validate = values => {
  const errors = {};
  const REQUIRED = 'Required';

  if(!values.email) {
    errors.email = REQUIRED;
  }else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)) {
    errors.email = 'Invalid email address';
  }

  if(!values.password) {
    errors.password = REQUIRED;
  }

  if(!values.name) {
    errors.name = REQUIRED;
  }

  if(!values.confirm) {
    errors.confirm = REQUIRED;
  }else if(values.confirm !== values.password) {
    errors.confirm = 'Passwords do not match';
  }

  return errors;
};

export const RegisterForm = props => {
  return (
    <form onSubmit={props.handleSubmit}>
      <H1>Register</H1>
      <Field name="email" type="text"
        component={RenderField} label="email"
      />
      <Field name="name" type="text"
        component={RenderField} label="name"
      />
      <Field name="password" type="password"
        component={RenderField} label="password"
      />
      <Field name="confirm" type="password"
        component={RenderField} label="confirm"
      />
      <Button submit>Register Now</Button>
    </form>
  );
};

RegisterForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
};

export default reduxForm({
  form: 'register',
  validate
})(RegisterForm);
