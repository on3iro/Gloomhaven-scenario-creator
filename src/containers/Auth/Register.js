/**
  * User registration page
  *
  * @namespace Register
  */

import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Redirect } from 'react-router-dom';

import * as selectors from './ducks/selectors';

import Wrapper from './Wrapper';
import RegisterForm from './RegisterForm';

import { registerSubmit } from './ducks/actions';


class Register extends Component {
  handleSubmit = values => {
    this.props.registerSubmit(values);
  }

  render() {
    const { from } = this.props.location.state || { from: { pathname: '/' } };

    return (
      this.props.isLoggedIn
        ? (
          <Redirect to={from} />
        ) : (
            <Wrapper>
              <RegisterForm onSubmit={this.handleSubmit} />
            </Wrapper>
        )
    );
  }
}

Register.propTypes = {
  registerSubmit: PropTypes.func.isRequired,
  location: PropTypes.object,
  isLoggedIn: PropTypes.bool,
};

const mapDispatchToProps = dispatch => {
  return bindActionCreators({
    registerSubmit,
  }, dispatch);
};

const mapStateToProps = (state) => {
  return {
    isLoggedIn: selectors.getLoggedIn(state),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Register);
