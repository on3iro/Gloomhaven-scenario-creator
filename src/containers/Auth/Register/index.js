/**
  * User registration page
  *
  * @namespace Register
  */

import React from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';

import * as selectors from '../ducks/selectors';

import Wrapper from '../Wrapper';
import RegisterForm from './RegisterForm';


const Register = props => {
  const { from } = props.location.state || { from: { pathname: '/' } };

  return (
    props.isLoggedIn
    ? (
      <Redirect to={from} />
    ) : (
      <Wrapper>
        <RegisterForm />
      </Wrapper>
    )
  );
}

const mapStateToProps = (state, ownProps) => {
  return {
    isLoggedIn: selectors.getLoggedIn(state),
  };
};

export default connect(mapStateToProps, null)(Register);
