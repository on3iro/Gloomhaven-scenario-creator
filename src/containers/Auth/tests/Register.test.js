import React from 'react';
import { shallow } from 'enzyme';
import { Redirect } from 'react-router-dom';

import Wrapper from '../Wrapper';
import { Register } from '../Register';
import RegisterForm from '../RegisterForm';


const renderWrapper = newProps => {
  const props = {
    registerSubmit: jest.fn(),
    location: {
      state: 'test'
    },
    isLoggedIn: false,
    ...newProps,
  };

  const renderedWrapper = shallow(<Register {...props} />);

  return {
    props,
    renderedWrapper,
  };
}

describe('<Register />', () => {
  it('should render <Wrapper />', () => {
    const { renderedWrapper } = renderWrapper();
    expect(renderedWrapper.find(Wrapper).length).toMatchSnapshot();
  });

  it('should render <RegisterForm />', () => {
    const { renderedWrapper } = renderWrapper();
    expect(renderedWrapper.find(RegisterForm).length).toMatchSnapshot();
  });

  it('should render <Redirect /> if user is logged in', () => {
    const { renderedWrapper } = renderWrapper({ isLoggedIn: true });
    expect(renderedWrapper.find(Redirect).length).toMatchSnapshot();
  });

  it('should call registerSubmit() onSubmit', () => {
    const { renderedWrapper, props } = renderWrapper();

    renderedWrapper.instance().handleSubmit('test');

    expect(props.registerSubmit).toHaveBeenCalledWith('test');
  });
});
