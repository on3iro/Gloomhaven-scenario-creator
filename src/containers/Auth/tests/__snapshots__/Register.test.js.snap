exports[`<Register /> should render <Redirect /> if user is logged in 1`] = `1`;

exports[`<Register /> should render <RegisterForm /> 1`] = `1`;

exports[`<Register /> should render <Wrapper /> 1`] = `1`;
